src/components/admin/hotels/manage-hotels.js

```diff
@@ -1,10 +1,72 @@
 import React from 'react';
-import Typography from '@material-ui/core/Typography';
-
-export default function ManageHotels() {
-  return (
-    <Typography paragraph>
-      The 'Manage Hotels' Page comming soon...
-    </Typography>
-  );
-}
\ No newline at end of file
+import {DataGrid} from '@material-ui/data-grid'
+import NoItem from '../../common/no-item';
+import SearchPlugin from '../../common/search-plugin';
+import {seedData, columns} from './manage-hotels-constants';
+
+class ManageHotels extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            hotelsRows: [],
+            filteredList: [],
+            searchTerm: ''
+        };
+    }
+
+    componentDidMount() {
+        // getting data from API
+        const hotelsRows = seedData;
+
+        this.setState({
+            hotelsRows: hotelsRows,
+            filteredList: hotelsRows
+        });
+    }
+
+    filterList = (searchTherm) => {
+        const term = searchTherm.toLowerCase();
+
+        const filteredList = this.state.hotelsRows.filter(e =>
+            e.name.toLowerCase().includes(term) ||
+            e.address.toLowerCase().includes(term) ||
+            e.contactName.toLowerCase().includes((term)));
+
+        this.setState({filteredList});
+    };
+
+    render() {
+        return (
+            this.state.hotelsRows.length === 0 ?
+                <NoItem
+                    singularItemName={'hotel'}
+                    pluralItemName={'hotels'}
+                    addNewItem={() => console.log('Added new item.')}/>

```

Comment:
Never pass arrow function as a callback. It will create and pass a new function after each render